MapboxNavigation/NavigationMapView.swift

```diff
@@ -199,6 +199,17 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         }
     }
     
+    /**
+     Resets all properties on `MGLMapView` around user course tracking.
+     
+     If reusing the map view outside of navigation, call this before attempting to change `MGLMapView.userTrackingMode` or changing `MGLMapView.showsUserLocation`.
+     */
+    @objc public func disableUseCourseTracking() {
```

Comment:
This one-time function feels kludgy to me. Instead, NavigationMapView should do a better job of distinguishing between course tracking and location/heading tracking when it comes to the `userTrackingMode` property. Outside of turn-by-turn navigation, there’s no need to indicate the user’s course per se.

Reply:
I agree. The problem is that `tracksUserCourse = false` means two different things depending on your "mode". If you're in navigation made and `tracksUserCourse == false` this means you have moved the map away from the puck (and you get the smaller puck). When not navigating, `tracksUserCourse == false` IMO means that the custom puck view should be hidden.

Reply:
In other words, `tracksUserCourse` is a bit of a misnomer; it should really be called something like `pointOfView` (first-person versus third-person) or `isOverhead`. `userTrackingMode` is kind of vestigial inside RouteMapViewController, since the map SDK only adjusts the center coordinate and heading but not the pitch and zoom level.

> When not navigating, `tracksUserCourse == false` IMO means that the custom puck view should be hidden.

Let’s make that explicit: a new `showsUserCourse` property that controls whether UserCourseView is visible. We can make it mutually exclusive of `showsUserLocation` to keep the map and navigation SDKs from both displaying an indicator simultaneously. What’s nice is that the developer can explicitly show the course view without following the user’s course (which is what `MGLUserTrackingMode.followWithCourse` would kind of do).